woocommerce/woocommerce
# Add locale param to jetpack redirect url

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds `locale=:site_locale_first_two_letters` to the Jetpack auth URL, allowing users to see their site language on the Jetpack auth pages.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Do not proceed with the core profiler yet.
3. Copy the JN site URL and open it on an incognito browser (you must be logged out from WPCOM)
4. Go to /wp-admin and click on `Settings -> General`
5. Change `Site Language`
6. Start the core profiler and make sure to select `Jetpack`
7. Continue to the Jetpack auth package.
8. Confirm the language change.

Jetpack auth page with `locale=es`
![Screenshot 2024-09-13 at 5 55 36 PM](https://github.com/user-attachments/assets/7429e5c0-46b8-4c94-ae8c-3caa4f558aa7)



Known issue: When you click on a link from the Jetpack auth page, it does not carry over `locale` param. This is something we need to update on wp-calypso separately.

For the logged-in sessions, WPCOM always uses the language from WPCOM's account setting regardless of the `locale` value.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add `locale` param when redirecting to the Jetpack auth page.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
